com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java

```diff
@@ -490,11 +499,20 @@ public static TextureSetResult createTextureSet(List<TextureSetLayout.Layout> la
 
         allRects.sort(Comparator.comparing(o -> o.getIndex()));
 
-        Pair<TextureSet.Builder, List<UVTransform>> vertexData = buildData(layoutWidth, layoutHeight, allRects, iterator);
+        // Assuming that the first texture is the largest
+        int largestPageWidth = layouts.get(0).getWidth();
+        int largestPageHeight = layouts.get(0).getHeight();
+
+        Pair<TextureSet.Builder, List<UVTransform>> vertexData = buildData(largestPageWidth, largestPageHeight, allRects, iterator);
         TextureSet.Builder builder = vertexData.left;
         for (Rect rect : allRects) {
+            // Since the actual textures may be of different size (in the editor), we'll use the correct page size
+            Layout layout = layouts.get(rect.getPage());
+            int pageWidth = layout.getWidth();
+            int pageHeight = layout.getHeight();
```

Comment:
The editor uses multiple textures, and as they may be of different size (from texture Packer), we make sure at least the calculated UV's are correct (for both editor and runtime)